test/e2e/framework/statefulset/fixtures.go

```diff
@@ -111,6 +133,34 @@ func NewStatefulSetPVC(name string) v1.PersistentVolumeClaim {
 	}
 }
 
+// NewStatefulSetPVCWithStorageClass returns a PersistentVolumeClaim named name, with specified storageclass
+// for testing StatefulSets.
+func NewStatefulSetPVCWithStorageClass(name, storageclass string) v1.PersistentVolumeClaim {
+	claim := NewStatefulSetPVC(name)
+	claim.Spec.StorageClassName = &storageclass
+	framework.Logf("claim pod %v", claim)
+	return claim
+}
+
+// NewStorageClassForWindows returns a StorageClass for windows with fsType set to ntfs, and provisioner
+// is set to gce-pd. The annotation indicates it is a storageclass for Windows too.
+func NewStorageClassForWindows(c clientset.Interface, namespace string) *storagev1.StorageClass {
+	if !framework.ProviderIs("gce", "gke") {
+		return nil
+	}
+	provisioner := "kubernetes.io/gce-pd"
```

Comment:
I don't think we should be putting provider-specific logic into the statefulset test. It will not be extensible for other providers. Can we instead install a Windows compatible storageclass in https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/storage-class/gce?

Reply:
Yes, I am working on it.